Packages/KeyAppKit/Sources/KeyAppKitCore/Async/Value/AsyncValue.swift

```diff
@@ -156,8 +156,13 @@ public class AsyncValue<T> {
 
             // Initialising failure
             if state.status == .initializing, error != nil, value == nil {
-                state.status = .initializing
-                stateSubject.send(state)
+                if error != nil {
+                    state.status = .ready
+                    stateSubject.send(state)
+                } else {
+                    state.status = .initializing
```

Comment:
will this condition work? if we inside 'error != nil' if

Reply:
we might need smth like if error != nil {} else if state.status == .initializing, value == nil { } else { } or this is nonsense? :)

Reply:
Not sure I got your point